CRM/Core/Payment/PayPalIPN.php

```diff
@@ -336,7 +336,8 @@ public function main() {
 
     Civi::log()->debug('PayPalIPN: Received (ContactID: ' . $ids['contact'] . '; trxn_id: ' . $input['trxn_id'] . ').');
 
-    if ($this->retrieve('membershipID', 'Integer', FALSE)) {
+    // Debugging related to possible missing membership linkage
+    if ($contributionRecurID && $this->retrieve('membershipID', 'Integer', FALSE)) {
```

Comment:
Doesn't this stop the membership processing from running if we don't have a recurring contribution?  Where I found this Paypal was passing in a contribution ID + membership ID but it was not an auto-renew so there was no recurring.  But the IPN still needed processing to "complete" the contribution and activate the membership.

Reply:
@mattwire I don't think so - this block was added to attempt to add in membership id if the underlying data had an issue to get back to behaviour that existed prior to one of the membership cleanups we collaborated on https://github.com/totten/civicrm-core/commit/fefee63643af32bc28fdc43196e886ede6194006#diff-508afec319247dd2f89dea484e473281R339 - skipping the block should be OK